cart-service/src/main/java/com/petclinic/cartsservice/businesslayer/CartServiceImpl.java

```diff
@@ -168,6 +171,85 @@ public Mono<CartResponseModel> createNewCart(CartRequestModel cartRequestModel)
         return cartRequestModelMono;
     }
 
+    @Override
+    public Mono<CartResponseModel> addProductToCart(String cartId, String productId, int quantity) {
```

Comment:
Help me understand what you mean by quantity. quantity of what?

Reply:
By reading the function i ended up understanding. It is very well done. maybe change the name of quantity by productQuantityAvailable to not get confused with the quantity in cart